tools/validators/instance_validator/validate/entity_instance.py

```diff
@@ -60,6 +62,69 @@ def _ValidateType(self):
 
     return True
 
+  def _ValidateTranslation(self):
+    """Uses information from the generated ontology universe to validate
+    an entity's translation if it exists.
+
+    Returns:
+      Returns boolean for validity of entity translation, defaults to True if
+      translation is not specified.
+    """
+    if self.translation_key not in self.entity.keys():
+      return True
+
+    translation_body = self.entity[self.translation_key]
+    if isinstance(translation_body.data, str):
+      if translation_body.data == self.translation_compliant:
+        return True
+      else:
+        print('Invalid translation compliance string', translation_body.data)
+        return False
+
+    # iterate through each translation device key and determine its form
+    for device_name in translation_body.keys():
+      # check if keys are UDMI compliant
+      if isinstance(translation_body[device_name].data, str):
+        continue
+
+      device_map = translation_body[device_name]
```

Comment:
use this above before the isinstance so you can reuse it

Reply:
Sometimes `translation_body` can be the string `COMPLIANT` so I need to first check for that before treating it as a Python dictionary.